### PR TITLE
Add schemas for Azure Front Door

### PR DESCRIPTION
**Pull request by jmelvinwork**
Add schemas for Azure Front Door

### CODE REVIEW

File: schemas/2015-01-01/deploymentTemplate.json [hunk 1]
```diff
@@ -51,6 +51,8 @@
               },
               {
                 "oneOf": [
+                 { "$ref": "https://schema.management.azure.com/schemas/2018-08-01/Microsoft.Network.json#/resourceDefinitions/frontDoors" },
```
Comment (TianoMS) [inline schemas/2015-01-01/deploymentTemplate.json]: >  [](start = 16, length = 1)

align

File: schemas/2018-08-01/Microsoft.Network.json [hunk 2]
```diff
@@ -4,6 +4,163 @@
...
   "resourceDefinitions": {
+    "frontDoors": {
+      "name": "string",
+      "type": "Microsoft.Network/frontDoors",
+      "apiVersion": "2018-08-01",
+      "location": "string",
+      "tags": {},
+      "properties": {
+        "friendlyName": "string",
+        "routingRules": [
```
Comment (TianoMS) [inline schemas/2018-08-01/Microsoft.Network.json]: >routingRules [](start = 9, length = 12)

are these manually written?

File: schemas/2018-08-01/Microsoft.Network.json [hunk 3]
```diff
@@ -4,6 +4,163 @@
   "title": "Microsoft.Network",
   "description": "Microsoft Network Resource Types",
   "resourceDefinitions": {
+    "frontDoors": {
```
Comment (TianoMS) [inline schemas/2018-08-01/Microsoft.Network.json]: >frontDoors [](start = 5, length = 10)

please add tests for the new resource types in tests/2018-08-01.

File: schemas/2018-08-01/Microsoft.Network.json (hunk 2 above)
Comment (jmelvinwork) [reply schemas/2018-08-01/Microsoft.Network.json]: I used the autorest tool to generate the schemas.

File: schemas/2018-08-01/Microsoft.Network.json (hunk 3 above)
Comment (jmelvinwork) [reply schemas/2018-08-01/Microsoft.Network.json]: I added some minimal tests.

**Comment by Tiano2017** [comment]
there are some conflicts. please sync and resolve.

**Comment by TianoMS** [comment]
> =======

resolve..

---
Refers to: schemas/2014-04-01-preview/deploymentTemplate.json:2173 in 0799e3a. [](commit_id = 0799e3a05df2a6a32008a601003bb528bfaed12f, deletion_comment = False)

**Comment by TianoMS** [comment]
>                   },

Let's not format this file as it has too many lines. You may just follow the current format. 

---
Refers to: schemas/2014-04-01-preview/deploymentTemplate.json:2169 in 0799e3a. [](commit_id = 0799e3a05df2a6a32008a601003bb528bfaed12f, deletion_comment = False)

**Comment by TianoMS** [comment]
>                   },

same. let's not format this file. 

---
Refers to: schemas/2015-01-01/deploymentTemplate.json:2177 in 0799e3a. [](commit_id = 0799e3a05df2a6a32008a601003bb528bfaed12f, deletion_comment = False)

File: tests/2018-08-01/Microsoft.Network.FrontDoor.tests.json [hunk 4]
```diff
@@ -0,0 +1,57 @@
+{
+  "tests": [
+    {
+      "name": "front door tests - minimal resource",
```
Comment (TianoMS) [inline tests/2018-08-01/Microsoft.Network.FrontDoor.tests.json]: >"name": "front door tests - minimal resource", [](start = 6, length = 46)

this is good enough.

**Comment by jmelvinwork** [comment]
> > =======
> 
> resolve..
> 
> Refers to: schemas/2014-04-01-preview/deploymentTemplate.json:2173 in [0799e3a](https://github.com/Azure/azure-resource-manager-schemas/commit/0799e3a05df2a6a32008a601003bb528bfaed12f). [](commit_id = [0799e3a](https://github.com/Azure/azure-resource-manager-schemas/commit/0799e3a05df2a6a32008a601003bb528bfaed12f), deletion_comment = False)

Resolved

**Comment by jmelvinwork** [comment]
> > ```
> >               },
> > ```
> 
> same. let's not format this file.
> 
> Refers to: schemas/2015-01-01/deploymentTemplate.json:2177 in [0799e3a](https://github.com/Azure/azure-resource-manager-schemas/commit/0799e3a05df2a6a32008a601003bb528bfaed12f). [](commit_id = [0799e3a](https://github.com/Azure/azure-resource-manager-schemas/commit/0799e3a05df2a6a32008a601003bb528bfaed12f), deletion_comment = False)

I reverted the formatting changes

File: schemas/2018-08-01/Microsoft.Network.json [hunk 5]
```diff
@@ -43,51 +52,112 @@
...
+      },
+      "required": [
+        "name",
+        "type",
+        "apiVersion",
+        "properties"
+      ],
+      "description": "Microsoft.Network/frontDoors"
+    },
+    "frontDoors_backendPools": {
```
Comment (TianoMS) [inline schemas/2018-08-01/Microsoft.Network.json]: >frontDoors_backendPools [](start = 5, length = 23)

this resource type is not in the deploymentTemplate.json. Is it expected?

File: schemas/2018-08-01/Microsoft.Network.json [hunk 6]
```diff
@@ -43,51 +52,112 @@
...
         "type",
         "apiVersion",
-        "location",
         "properties"
       ],
-      "description": "Microsoft.Network/applicationGateways"
+      "description": "Microsoft.Network/frontDoors/backendPools"
     },
-    "applicationSecurityGroups": {
+    "frontDoors_healthProbeSettings": {
```
Comment (TianoMS) [inline schemas/2018-08-01/Microsoft.Network.json]: >frontDoors_healthProbeSettings [](start = 5, length = 30)

these resource types too.

**Pull request merged**